basic_nodes/src/main/java/club/mondaylunch/gatos/basicnodes/GetAtIndexNodeType.java

```diff
@@ -0,0 +1,62 @@
+package club.mondaylunch.gatos.basicnodes;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import club.mondaylunch.gatos.core.data.DataBox;
+import club.mondaylunch.gatos.core.data.DataType;
+import club.mondaylunch.gatos.core.data.ListDataType;
+import club.mondaylunch.gatos.core.data.OptionalDataType;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector.Input;
+import club.mondaylunch.gatos.core.graph.connector.NodeConnector.Output;
+import club.mondaylunch.gatos.core.graph.type.NodeType;
+
+public class GetAtIndexNodeType extends NodeType.Process {
+    @Override
+    public Map<String, DataBox<?>> settings() {
+        return Map.of();
+    }
+
+    @Override
+    public Set<Input<?>> inputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        return Set.of(
+            new NodeConnector.Input<>(nodeId, "input", ListDataType.GENERIC_LIST),
+            new NodeConnector.Input<>(nodeId, "index", DataType.NUMBER)
+        );
+    }
+
+    @Override
+    public Set<Output<?>> outputs(UUID nodeId, Map<String, DataBox<?>> settings, Map<String, DataType<?>> inputTypes) {
+        var inputType = inputTypes.getOrDefault("input_type", OptionalDataType.GENERIC_OPTIONAL);
```

Comment:
? There is no input called 'input_type' or one that takes an optional - this won't work.

Reply:
this has been resolved